app/code/Magento/InventoryConfiguration/Model/StockConfigurationValidator.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: dev
+ * Date: 08.02.18
+ * Time: 23:54
+ */
+
+namespace Magento\InventoryConfiguration\Model;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Catalog\Model\ResourceModel\Product as ProductResourceModel;
+use Magento\CatalogInventory\Model\Configuration;
+use Magento\CatalogInventory\Model\Stock\StockItemRepository as LegacyStockItemRepository;
+use Magento\CatalogInventory\Model\Stock\Item as LegacyStockItem;
+use Magento\CatalogInventory\Api\Data\StockItemInterface;
+use Magento\CatalogInventory\Api\StockItemCriteriaInterfaceFactory;
+
+class StockConfigurationValidator implements StockConfigurationInterface
+{
+    /**
+     * @var StockConfigurationInterface[]
+     */
+    protected $validators;
+
+    /**
+     * @var StockItemCriteriaInterfaceFactory
+     */
+    private $stockItemCriteriaFactory;
+
+    /**
+     * @var LegacyStockItemRepository
+     */
+    protected $legacyStockItemRepository;
+
+    /**
+     * @var Configuration
+     */
+    protected $configuration;
+
+    /**
+     * @var ProductResourceModel
+     */
+    protected $productResource;
+
+    /**
+     * StockValidationComposer constructor.
+     * @param StockConfigurationInterface[] $validators
+     * @param Configuration $configuration
+     * @param LegacyStockItemRepository $legacyStockItemRepository
+     * @param StockItemCriteriaInterfaceFactory $stockItemCriteriaFactory
+     * @param ProductResourceModel $productResource
+     * @throws LocalizedException
+     */
+    public function __construct(
+        array $validators=[],
+        Configuration $configuration,
+        LegacyStockItemRepository $legacyStockItemRepository,
+        StockItemCriteriaInterfaceFactory $stockItemCriteriaFactory,
+        ProductResourceModel $productResource
+    ) {
+        $this->configuration = $configuration;
+        $this->legacyStockItemRepository = $legacyStockItemRepository;
+        $this->productResource = $productResource;
```

Comment:
you need to use 
```

namespace Magento\InventoryCatalog\Model;

/**
 * Provides all product SKUs by ProductIds. Key is sku, value is product id
 * @api
 */
interface GetProductIdsBySkusInterface
{
    /**
     * @param array $skus
     * @return array
     */
    public function execute(array $skus): array;
}
```
instead of Magento\Catalog\Model\ResourceModel\Product

Reply:
Done